### DISCUSSION TITLE
Publish to Codeberg

### DISCUSSION THREAD

**Discussion by SHogenboom** [quarto-dev/quarto-cli#6737] [Feature Requests]
### Description

I want to publish a Quarto Book on Codeberg (an opensource git client). The current workflow - which works but is inconvenient:

1. Render the book locally (default)
2. `quarto publish gh-pages` >>> creates the output files on a branch called `gh-pages`
3. rename the branch `gh-pages` to `pages` (the required branchname for Codeberg Pages)

What would be ideal if one could specify the name of the target branch when using the publish command. For example:
`quarto publish gh-pages —branch pages` or something of the likes. The `gh-pages` can then still be used to trigger the correct workflow, but the output book is added to a differently named branch.

Thanks in advance if you can make this work!

**Comment by cscheid** [comment]
> Codeberg (an opensource git client).

I think you have some misconceptions. Codeberg is not a git client. It's a different website (different from GitHub, which I assume is what you're drawing your comparison from). `quarto publish gh-pages` does more than just push to a _git_ branch; it does a number of additional things that are specific to _GitHub_ (that have to do with authentication, etc). Supporting Codeberg would be a larger project.

I mention that because only because this is the first I (personally!) hear of Codeberg, and we'd want to know that there's a critical mass of users requesting this support.

**Reply by SHogenboom** [reply]
Thanks for the clarification. Because it works as described above, I hadn’t realised there were additional steps like authentication involved! Highly doubt Codeberg will take off anytime soon, researchers are still learning about version control to begin with ;) 

Will remove the feature request and deal with it manually.

**Comment by metazool** [comment]
Just to note that i'm here setting up quarto publication for my first time trying to start a new research project on Codeberg rather than Github. I'd _love_ the ability to do this in the `quarto` cli! 

I wonder how you could find (financial) support to introduce this alongside improvements / changes to the Codeberg Pages service that's now in "maintenance mode" https://codeberg.org/Codeberg/pages-server/issues/399
